### PR TITLE
Remove webpack 4 warnings

### LINKED ISSUE #62: support webpack4?
webpack4 has published, can u update this plugin for support webpack4?

### PR DESCRIPTION
**Pull request by ooflorent**
In webpack 4, `Tapable#apply` and `Tapable#plugin` raise a **deprecation notice**.

- `Tapable#apply(plugin)` must be replaced by a call to `plugin.apply`
- `Tapable#plugin(hook, fn)` must be replaced by a call to `hook.tap` (or `hook.tapAsync`)

Fix #62

### CODE REVIEW

**Comment by MartinX3** [comment]
Please merge this PullRequest.

**Comment by lerit** [comment]
please merge this pull,thank you

File: test/integration.spec.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ const webpackPromise = function (config, globalPlugins) {
   const compiler = webpack(config);
   compiler.outputFileSystem = new MemoryFileSystem();
   if (Array.isArray(globalPlugins)) {
-    globalPlugins.forEach(p => compiler.apply(p));
+    globalPlugins.forEach(p => p.apply(compiler));
```
Comment (christophehurpeau) [inline test/integration.spec.js]: why this change ?

Comment (ooflorent) [reply test/integration.spec.js]: `compiler.apply` is deprecated

Comment (christophehurpeau) [reply test/integration.spec.js]: does this still works in webpack 2 ? (which is the devDependency)

Comment (ooflorent) [reply test/integration.spec.js]: Yes

**Comment by chenxiaoqiang123** [comment]
I still have this problem at webpack 4.1.1, how can I deal it？

**Comment by sangdth** [comment]
When will this PR take effect if all checks have passed?

**Comment by lehni** [comment]
Until this gets merged, you can use the code from the PR by specifying this in your `package.json`:
```json
    "friendly-errors-webpack-plugin": "github:ooflorent/friendly-errors-webpack-plugin#98baeb9cd181ed0d8754e8a3afc4c4546516651c",
```

**Comment by sangdth** [comment]
Thanks, @lehni , I followed your tip and fix 999+ bugs in my code.

**Comment by timneutkens** [comment]
Released as friendly-errors-webpack-plugin@1.7.0
